Wikipedia/UI-V5/WMFNearbySectionController.m

```diff
@@ -79,8 +82,8 @@ - (NSString*)footerText {
 }
 
 - (NSArray*)items {
-    if ([self.nearbyResults.results count] > 0) {
-        return self.nearbyResults.results;
+    if ([self.viewModel.locationSearchResults.results count] > 0) {
```

Comment:
@bgerstle Yeah… i think you are right - covering them up necessarily fixing much.

This really about the code smell - inappropriate intimacy. 

Here I think its a clue about the abstraction not quite bing right. Why does this object need to reach that far down to get it info it needs to do its job? 

One thought may be that the NearbySectionController should defer *more* responsibility to the "view model". Like the nearby section controller should be even a thinner veneer over the view model. 

Another thought is that maybe the section controller should be implemented by a subclass of the view model or implemented by the view model itself.